xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -97,11 +97,19 @@ void CGUIWindowPVRGuide::StopRefreshTimelineItemsThread()
 
 void CGUIWindowPVRGuide::RegisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
   g_EpgContainer.RegisterObserver(this);
+  if (g_PVRTimers)
+    g_PVRTimers->RegisterObserver(this);
+  CGUIWindowPVRBase::RegisterObservers();
 }
 
 void CGUIWindowPVRGuide::UnregisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
+  CGUIWindowPVRBase::UnregisterObservers();
+  if (g_PVRTimers)
```

Comment:
> PVRManager instance is owned by ServiceManager which is owned by CApplication. A restart of PVRManager won't create a new instance.

 yeah. looked this up myself meanwhile. good that you confirm. so, no lifecycle issues when using g_PVRManager as proxy for messaging.

using window messages as replacement for current observer approach relies on the fact that there can only be max one active window at a time, right? messaging is 1:1, observer is 1:n communication, right? So, what if currently there are non-window observers (have not checked this yet) the messaging approach wont work, no.